### PR TITLE
hierarchical module resolution when looking for jest/package.json

### PR DESCRIPTION
**Pull request by ronp001**
Add support for setups in which jest is installed in an ancestor directory.

(As per [node](https://nodejs.org/api/modules.html#modules_loading_from_node_modules_folders) docs, the jest module can be located in a `node_modules` directory anywhere upwards in the directory hierarchy).

Note that this also relates to #218 (as inability to resolve jest version might be due to such a hierarchical setup).

### CODE REVIEW

File: src/helpers.ts [hunk 1]
```diff
@@ -56,13 +56,18 @@ export function pathToConfig(pluginSettings: IPluginSettings) {
...
+  do {
+    for (const i in paths) {
+      const absolutePath = join(pluginSettings.rootPath, updir, paths[i])
+      if (existsSync(absolutePath)) {
+        return absolutePath
+      }
     }
-  }
+    updir += '../'
+  } while (join(pluginSettings.rootPath, updir) !== join(pluginSettings.rootPath, updir, '../'))
```
Comment (seanpoulter) [inline src/helpers.ts]: Unbounded loops make me nervous. Are there any cases where a shortcut / link could make this not end up at the root?

Comment (ronp001) [reply src/helpers.ts]: Good point.  Added a failsafe mechanism.

**Comment by ronp001** [comment]
I'm closing this PR because rhe code included here is incompatible with recent changes to master (which seem to have completely changed the logic which the plugin uses to look for package.json).

**Comment by seanpoulter** [comment]
I think they're complimentary changes @ronp001. The recent change shuffled things around but doesn't address the problem you're trying to solve. Let's get those changes switched around and ready to go. Want me to send you a PR?

**Comment by ronp001** [comment]
That will be great - thanks @seanpoulter.

**Comment by seanpoulter** [comment]
I had a look at this one over the weekend @ronp001. Does this fix a bug with the extension right now that using the `rootPath` or `pathToJest` settings doesn't cover? It seems to me like we're improving the "default" behavior a bit.

**Pull request closed without merging**